isarojdahal/learn.everydaykarmaa.com
# feat: Podomoro Timer

Implemented the 'Podomore timer' feature. This allows users to set a learning timer, which replaces the 'Start Learning' button once activated. When the timer reaches 0:00, a confetti animation is displayed in the center of the screen, celebrating the user's effort. The timer's state and settings are persistent, ensuring visibility across pages.

## Changes Made
- Added a 'Start Learning' (by default at the right bottom but it is draggable user can place it anywhere in the screen)
- Implemented a dialog box to prompt users for their preferred learning time.
- Replaced the 'Start Learning' button with a countdown timer when the user sets a duration.
- Added a confetti animation triggered when the timer reaches 0:00.
- Ensured the timer state is persistent across pages using local storage.

## How to Test
- Click the 'Start Learning' button.
- Enter the desired learning duration in the dialog box and confirm.
- Observe that the timer replaces the 'Start Learning' button and starts counting down.
- Navigate to different pages to confirm that the timer remains visible with the correct countdown.
- Wait for the timer to reach 0:00 and verify that a confetti animation appears in the center of the screen.
- Refreshing the page to check if it keep the timer state persistent


https://github.com/user-attachments/assets/26eac157-6d1d-467d-977b-98bc1d67770f
